src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt

```diff
@@ -85,3 +88,30 @@ fun mergeFlags(element: BoundElement<*>): TypeFlags =
 
 fun mergeFlags(tys: List<Ty>): TypeFlags =
     tys.fold(0) { a, b -> a or b.flags }
+
+tailrec fun Ty.isSized(): Boolean {
+    return when (this) {
+        is TyPrimitive,
+        is TyReference,
+        is TyPointer,
+        is TyArray,
+        is TyEnum,
+        is TyFunction -> true
+        is TySlice, is TyTraitObject -> false
+        is TyTypeParameter -> isSized
+        is TyStruct -> {
+            val namedFields = item.namedFields
+            val tupleFields = item.positionalFields
+            val typeRef = when {
+                namedFields.isNotEmpty() -> namedFields.last().typeReference
+                tupleFields.isNotEmpty() -> tupleFields.last().typeReference
+                else -> null
+            }
+            val type = typeRef?.type?.substitute(typeParameterValues) ?: return true
+            type.isSized()
+        }
+
+        is TyTuple -> types.last().isSized()
+        else -> false
```

Comment:
Is `TyNever` unhanded? 

I think we should return `true` by default here, because you have to opt-out of sizedness usually.

Reply:
> Is TyNever unhanded?

`TyNever` is primitive type so it is sized too)